### ISSUE TITLE
Bug: Redirect response requires media_type to be set to TEXT or HTML

### ISSUE THREAD

**Issue by danechitoaie** [litestar-org/litestar#1884]
### Description

If we return a redirect without setting any media type `Redirect("/?code=123", status_code=HTTP_307_TEMPORARY_REDIRECT)` the request will fail with 500 internal error due to `if media_type not in REDIRECT_ALLOWED_MEDIA_TYPES:` as 
```py
REDIRECT_ALLOWED_MEDIA_TYPES = {MediaType.TEXT, MediaType.HTML}
```

Any idea why do we care about the media type when doing a redirect?
And since JSON is the default media type any idea why at least JSON is not on that list?

The fix seems to be setting a media_type on the Response object to one of the allowed types.
```
```

### MCVE

```python
from litestar import Litestar, MediaType, get
from litestar.response import Redirect
from litestar.status_codes import HTTP_307_TEMPORARY_REDIRECT


@get("/")
async def index(code: str | None = None) -> Redirect | str:
    if not code:
        return Redirect("/?code=123", status_code=HTTP_307_TEMPORARY_REDIRECT)

    return "Hello, world!"


app = Litestar([index], debug=True)
```

### Steps to reproduce

```bash
1. Run with `uvicorn main:app`
2. Access `/`
3. Redirect will fail will 500 internal server error
```

### Litestar Version

2.0.0b2

### Platform

- [ ] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

<a href="https://polar.sh/litestar-org/litestar/issues/1884">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/1884/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/1884/pledge.svg">
</picture>
</a>

**Comment by cofin** [comment]
I couldn't think of a reason this should be disabled, so I've added it and it should be merged into `main` soon.  Thanks for reporting this!

**Comment by cofin** [comment]
@all-contributors, add @danechitoaie for bugs

**Comment by cofin** [comment]
Fixed by #1908
